test/dal/connection.spec.ts

```diff
@@ -0,0 +1,67 @@
+import {getConnectionDetails} from '../../src/dal/connection';
+import {expect} from 'chai';
+
+describe('DB connection logic', () => {
+  describe('getConnectionDetails()', () => {
+    const env = Object.assign({}, process.env);
+
+    after(() => {
+      process.env = env;
+    });
+
+    beforeEach(() => {
+      process.env = {
+        'DB_HOSTNAME': 'DB_HOSTNAME',
+        'DB_PORT': 'DB_PORT',
+        'DB_USERNAME': 'DB_USERNAME',
+        'DB_PASSWORD': 'DB_PASSWORD',
+        'DB_NAME': 'DB_NAME',
+      };
+    });
+
+    it('should get count of submissions', async () => {
+      const details = getConnectionDetails();
+      expect(details.host).to.equal('DB_HOSTNAME');
+      expect(details.port).to.equal('DB_PORT');
+      expect(details.user).to.equal('DB_USERNAME');
+      expect(details.password).to.equal('DB_PASSWORD');
+      expect(details.database).to.equal('DB_NAME');
+      expect(details.connectionLimit).to.equal(10);
```

Comment:
Probably a good idea to write tests for exceeding the connection limit.
Also, we should create constants for DB_CREDENTAILS_ERR, DB_CONNECTION_ERR; such constants that can be caught in the business logic, and thus show a friendly message to the user.

Reply:
This test just confirms our options contains the expected value, in case a change breaks it. I trust the module we are using to enforce it's options correctly.

Reply:
An error constant makes sense.